src/hooks/wallets/useOnboard.ts

```diff
@@ -67,19 +60,29 @@ export const getConnectedWallet = (wallets: WalletState[]): ConnectedWallet | nu
   }
 }
 
-const getWalletConnectLabel = async ({ label, provider }: ConnectedWallet): Promise<string | undefined> => {
-  if (label.toUpperCase() !== WALLET_KEYS.WALLETCONNECT.toUpperCase()) return
-
+const getWalletConnectLabel = async (wallet: ConnectedWallet): Promise<string | undefined> => {
   const UNKNOWN_PEER = 'Unknown'
-  const { default: WalletConnect } = await import('@walletconnect/client')
+  const { label } = wallet
+
+  const isWalletConnect = [
+    WALLET_KEYS.WALLETCONNECT.toUpperCase(),
+    WALLET_KEYS.WALLETCONNECT_V2.toUpperCase(),
+  ].includes(label.toUpperCase())
+
+  if (!isWalletConnect) return
 
-  const peerWallet =
-    ((provider as unknown as any).connector as InstanceType<typeof WalletConnect>).peerMeta?.name || UNKNOWN_PEER
+  const { connector } = wallet.provider as unknown as any
+  const peerWalletV2 = connector.session?.peer?.metadata?.name
+  const peerWalletV1 = connector.peerMeta?.name
 
-  return peerWallet ?? UNKNOWN_PEER
+  return peerWalletV2 || peerWalletV1 || UNKNOWN_PEER
 }
 
+let lastTracked = ''
 const trackWalletType = (wallet: ConnectedWallet) => {
+  if (wallet.label === lastTracked) return
+  lastTracked = wallet.label
```

Comment:
If the user disconnects and reconnects the same wallet, this will not track.

We could reset `lastTracked` on line 193 in an `else`. What do you think?

Reply:
Done.